### ISSUE TITLE
Simplify how users configure Sentry with Serilog

### ISSUE THREAD

**Issue by jamescrosswell** [getsentry/sentry-dotnet#2884]
Currently it's extremely confusing to configure Sentry to work with Serilog. See [this StackOverflow question](https://stackoverflow.com/questions/77454233/sentry-with-generic-host-and-serilog-net-7/77463866) to get some flavour, and also this issue/PR: 
- https://github.com/getsentry/sentry-dotnet/pull/2883 

Currently you can provide a Sentry DSN and various other settings via:
1. The `SentrySdk.Init` method in an application that's not using a Generic Host
2. The `SentryWebHostBuilderExtensions.UseSentry` extension method, when configuring the generic host
3. And/Or the `SentrySinkExtensions.Sentry` extension method when configuring Serilog

This is maybe an issue with the default value of the `initializeSdk` parameter that gets passed when initializing via Serilog. 

At the moment, this defaults to `null`:
https://github.com/getsentry/sentry-dotnet/blob/cf2d206612c5096efaf3f8e853ec4cfcf1fdbab0/src/Sentry.Serilog/SentrySinkExtensions.cs#L107

And that means whatever was in the `SentrySerilogOptions` before won't be overridden:
https://github.com/getsentry/sentry-dotnet/blob/cf2d206612c5096efaf3f8e853ec4cfcf1fdbab0/src/Sentry.Serilog/SentrySinkExtensions.cs#L302-L306

And with a shinny new set of `SentrySerilogOptions` that will be `true`:
https://github.com/getsentry/sentry-dotnet/blob/2285a7f3b805136ab858d4f61e456f30b8a1e2c4/src/Sentry.Serilog/SentrySerilogOptions.cs#L7-L12

So by default, if you initialize Sentry using `SentrySdk.Init` or `UsingSentry` and then you wire up Serilog to use Sentry via the `SentrySinkExtensions.Sentry` extension method we provide, without supplying any parameters to that method, you get an exception at runtime 😢  

## Brainstorming

One possible solution would be to detect, in the call to `SentrySinkExtensions.Sentry`, whether a call to initialize Sentry had already been made or not and, if so, use a default value of `false` for `initializeSdk`. 

Ordinarily, we could check SentrySdk.IsEnabled to work this out... but it turns out when using `SentryWebHostBuilderExtensions.UseSentry` initialization of the SDK is delayed:
https://github.com/getsentry/sentry-dotnet/blob/75843685494d04e0e955e2e48b40ca93d61aa8f6/src/Sentry.AspNetCore/SentryWebHostBuilderExtensions.cs#L79-L82

Potentially we need some other internal flag that we set in the static SentrySdk class before or after setting up the `ConfigureLogging` delegate in the call to `SentryWebHostBuilderExtensions.UseSentry` then.

**Comment by jamescrosswell** [comment]
Also see this comment on related PR: 
- https://github.com/getsentry/sentry-dotnet/pull/2883#issuecomment-1823511010

**Comment by kanadaj** [comment]
How about treating a null DSN for the Serilog call as equivalent to `initializeSdk: false`? I'd assume people generally don't initialize the SDK via the Serilog integration, and in the rare case they do they pass a Dsn. This is effectively eqiivalent to the current behaviour on 3.x I believe

**Comment by jamescrosswell** [comment]
> Though I guess this is basically equivalent to having multiple overloads

Yeah I think it amounts to the same thing... if you don't pass a DSN (and don't want to initialize Sentry) then there's no point in passing any of the other Sentry options - you'd specify those instead wherever you did initialize Sentry. In this case the value of "InitializeSdk" can be implicitly false, so no need to pass that either. 

The only question would be what to do with the overload that takes an `Action<SentrySerilogOptions>`. In that case, I think we need some reasonable default behaviour... maybe we leave that one as it is today (i.e. assume they do want to initialize the sdk).

**Comment by bitsandfoxes** [comment]
I'd rather work around this rather than break behaviour that has been in place like that for ages. So if we can resolve this by providing overloads in `WriteTo.Sentry()` all the more reason to do it.

I think it is fine to leave the overload that takes an `Action<SentrySerilogOptions>`  as it is today. If someone wants to really initialize the SDK through that.
